tests/integration/distribution/keeper/deterministic_test.go

```diff
@@ -0,0 +1,226 @@
+	require.NoError(t, stakingKeeper.Params.Set(newCtx, stakingtypes.DefaultParams()))
+
+	poolKeeper := poolkeeper.NewKeeper(cdc, runtime.NewEnvironment(runtime.NewKVStoreService(keys[pooltypes.StoreKey]), log.NewNopLogger()), accountKeeper, bankKeeper, stakingKeeper, authority.String())
+
+	distrKeeper := distrkeeper.NewKeeper(
+		cdc, runtime.NewEnvironment(runtime.NewKVStoreService(keys[distrtypes.StoreKey]), logger), accountKeeper, bankKeeper, stakingKeeper, poolKeeper, distrtypes.ModuleName, authority.String(),
+	)
+
+	authModule := auth.NewAppModule(cdc, accountKeeper, acctsModKeeper, authsims.RandomGenesisAccounts)
+	bankModule := bank.NewAppModule(cdc, bankKeeper, accountKeeper)
+	stakingModule := staking.NewAppModule(cdc, stakingKeeper, accountKeeper, bankKeeper)
+	distrModule := distribution.NewAppModule(cdc, distrKeeper, accountKeeper, bankKeeper, stakingKeeper, poolKeeper)
+	poolModule := protocolpool.NewAppModule(cdc, poolKeeper, accountKeeper, bankKeeper)
+
+	addr := sdk.AccAddress(PKS[0].Address())
+	valAddr := sdk.ValAddress(addr)
+	valConsAddr := sdk.ConsAddress(valConsPk0.Address())
+
+	// set proposer and vote infos
+	ctx := newCtx.WithProposer(valConsAddr).WithCometInfo(comet.Info{
+		LastCommit: comet.CommitInfo{
+			Votes: []comet.VoteInfo{
+				{
+					Validator: comet.Validator{
+						Address: valAddr,
+						Power:   100,
+					},
+					BlockIDFlag: comet.BlockIDFlagCommit,
+				},
+			},
+		},
+	})
+
+	integrationApp := integration.NewIntegrationApp(ctx, logger, keys, cdc,
+		encodingCfg.InterfaceRegistry.SigningContext().AddressCodec(),
+		encodingCfg.InterfaceRegistry.SigningContext().ValidatorAddressCodec(),
+		map[string]appmodule.AppModule{
+			authtypes.ModuleName:    authModule,
+			banktypes.ModuleName:    bankModule,
+			stakingtypes.ModuleName: stakingModule,
+			distrtypes.ModuleName:   distrModule,
+			pooltypes.ModuleName:    poolModule,
+		},
+		msgRouter,
+		grpcRouter,
+	)
+
+	sdkCtx := sdk.UnwrapSDKContext(integrationApp.Context())
+
+	// Register MsgServer and QueryServer
+	distrtypes.RegisterMsgServer(integrationApp.MsgServiceRouter(), distrkeeper.NewMsgServerImpl(distrKeeper))
+	distrtypes.RegisterQueryServer(integrationApp.QueryHelper(), distrkeeper.NewQuerier(distrKeeper))
+
+	qr := integrationApp.QueryHelper()
+	queryClient := distrtypes.NewQueryClient(qr)
+
+	return &deterministicFixture{
+		app:           integrationApp,
+		sdkCtx:        sdkCtx,
+		cdc:           cdc,
+		keys:          keys,
+		accountKeeper: accountKeeper,
+		bankKeeper:    bankKeeper,
+		distrKeeper:   distrKeeper,
+		stakingKeeper: stakingKeeper,
+		poolKeeper:    poolKeeper,
+		addr:          addr,
+		valAddr:       valAddr,
+		queryClient:   queryClient,
+	}
+}
+
+func TestQueryParamsDeterministic(t *testing.T) {
```

Comment:
@tac0turtle can you have a look at this, too see if i'm going the right direction.?